pkg/security/oauth2/auth/authorize/middleware.go

```
@@ -163,13 +170,17 @@ func (mw *AuthorizeEndpointMiddleware) ApproveOrDenyHandlerFunc() gin.HandlerFun
 		logger.WithContext(ctx).Debug(fmt.Sprintf("AuthorizeRequest: %s", request))
 
 		// parse approval params and check
-		approval := mw.parseApproval(ctx)
+		approval, remember := mw.parseApproval(ctx)
 		if e := auth.ValidateApproval(ctx, approval, client, request.Scopes); e != nil {
 			mw.handleError(ctx, e)
 			return
 		}
 		request.Approved = true
 
+		if remember {
+			_ = mw.saveApprovedRequest(ctx, request)
+		}
+
```

error is ignored here, maybe add a log